snapcraft/application.py

```diff
@@ -155,7 +168,10 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
         try:
             return super()._resolve_project_path(project_dir / "snap")
         except FileNotFoundError:
-            return super()._resolve_project_path(project_dir)
+            try:
```

Comment:
At some point (probably not this PR), we should refactor the nested try-excepts and raise a descriptive error message.  

Otherwise, they will get a FileNotFound for `build-aux/snap/snapcraft.yaml`, which may be confusing.

Reply:
Agreed - also that error should probably be a form of `CraftError` when it comes out of this since we don't want it to me marked as an internal error.

Reply:
Kicking the can down the road: https://github.com/canonical/snapcraft/issues/4636

I've placed this on next pulse, but I may have time to do it today.